pkg/bsql/user_syncer.go

```diff
@@ -136,6 +136,21 @@ func (s *userSyncer) CreateAccount(
 		return nil, nil, nil, err
 	}
 
+	previousAccountResource, err := s.validateAccount(ctx, provisioningConfig, queryInputs)
+	if err != nil {
+		if !errors.Is(err, ErrUnableFindResourceProvisioning) {
+			return nil, nil, nil, fmt.Errorf("failed to validate created account: %w", err)
+		}
+	}
+
+	if previousAccountResource != nil {
+		logger.Info("account resource is already created", zap.String("resource_id", previousAccountResource.GetId().GetResource()))
```

Comment:
If it's already created, don't we want to update the password based on the credential options?